src/Juvix/Core/IR/Types.hs

```diff
@@ -66,7 +147,8 @@ data Name
 data Value primTy primVal m
```

Comment:
Do we want to either add extensions to values (so we can pass them through evaluation), or alter evaluation so it just returns a suitably-restricted `Term` (though we couldn't typecheck this)? We will want the ability to evaluate but keep annotations (maybe with some transform).

Reply:
I think we probably will at some point, yes. My reasoning for stripping the extensions in `eval*` was that I thought they would most likely be annotations (e.g. types) that we may want to recalculate to make sure the evaluator works how we expect.